src/systems/engine.cppm

```diff
@@ -31,4 +37,22 @@ export namespace ecs::systems::engine {
         if (display->get().window->is_input_active(abstractions::gui::inputs::down))
             pos.y += c.speed;
     }
+
+    constexpr void update_hitbox_position(ecs::components::position &pos, ecs::components::engine::hitbox &box) noexcept {
+        box.box.x = pos.x;
+        box.box.y = pos.y;
+    }
+
+    constexpr void collision(entity e, entity_container &ec, components::engine::hitbox &box) noexcept
+    {
+        std::ranges::for_each(ec.get_entities(), [&](entity other) {
+            if (e == other)
+                return;
+            auto other_box = ec.get_entity_component<components::engine::hitbox>(other);
+            if (other_box.has_value() && box.box.intersects(other_box->get().box) && !box.is_trigger && !other_box->get().is_trigger) {
```

Comment:
I suppose `!box.is_trigger && !other_box->get().is_trigger` is meant to prevent a hitbox from triggering other hitbox if it's already triggered, but that prevents a triple collision from triggering one arbitrary hitbox of the three.
Makes some sense if a collision would unconditionally erase the entities, but as the game engine, that decision is not up to this particular code.